Ghidra/Framework/SoftwareModeling/src/main/java/ghidra/app/plugin/assembler/sleigh/SleighAssembler.java

```diff
@@ -236,6 +243,38 @@ public AssemblyResolutionResults resolveLine(Address at, String line, AssemblyPa
 		}
 		return results;
 	}
+	
+	@Override
+	public AssemblyResolutionResults resolveLine(Address at, InstructionNode line,
+			AssemblyPatternBlock ctx, TaskMonitor monitor) throws AssemblySyntaxException {
+
+		if (!ctx.isFullMask()) {
+			throw new AssemblyError(
+				"Context must be fully-specified (full length, no shift, no unknowns)");
+		}
+		if (lang.getContextBaseRegister() != null &&
+			ctx.length() < lang.getContextBaseRegister().getMinimumByteSize()) {
+			throw new AssemblyError(
+				"Context must be fully-specified (full length, no shift, no unknowns)");
+		}
+		Collection<AssemblyParseResult> parse = parseLine(line, monitor);
+		if (monitor.isCancelled()) {
+			return new AssemblyResolutionResults();
```

Comment:
I don't remember our exact reasoning here. At one point it was handy to keep the partial results because of how slow it was, but now I don't think that's as big a deal (it's faster now due to other changes). It seems we'd have to add a `throws CancelledException` if we want to go with that (cleaner) style check. I don't have a big preference either way.